Core/Src/stm32f4xx_it.c

```diff
@@ -220,28 +240,17 @@ void DMA2_Stream0_IRQHandler(void)
 void CAN2_RX0_IRQHandler(void)
 {
   /* USER CODE BEGIN CAN2_RX0_IRQn 0 */
-  can_receive_callback(&hcan2);
+  //can_receive_callback(&hcan2);
+
+  /* If ANY CAN msg is received on CAN 2, we can assume it is on charger cart */
+  compute_set_charger_connected();

```

Comment:
Note the first can message sent is **after** the charger is enabled so we have no way of knowing charge current until after the car starts. I can change this via TPU if needed by making charger box TPU send a message. Kinda scary imo.

Reply:
not sure i understand your point here